atlasdb-config/src/main/java/com/palantir/atlasdb/factory/ServiceDiscoveringAtlasSupplier.java

```diff
@@ -73,6 +79,35 @@ public KeyValueService getKeyValueService() {
         return keyValueService.get();
     }
 
+    private KeyValueService pollOrCreateRawKeyValueService(AtlasDbFactory atlasFactory) {
+        return atlasDbConfig.pollForDataBaseOnStartup() ? pollForKeyValueService(atlasFactory)
+                : atlasFactory.createRawKeyValueService(keyValueServiceConfig, leaderConfig);
+    }
+
+    private KeyValueService pollForKeyValueService(AtlasDbFactory atlasFactory) {
+        int failureCount = 0;
```

Comment:
We probably want to move this logic into a helper class. Also, we could consider using `Awaitility` as the implementation.